File: cache/write_cache.py
# Copyright 2018 BACnet Gateway.  All rights reserved.

import argparse
import cache_db

# Get command line arguments
parser = argparse.ArgumentParser( description='Maintain cache of recent values used by Building Monitor', add_help=False )
parser.add_argument( '-w', dest='view' )
parser.add_argument( '-f', dest='facility' )
parser.add_argument( '-i', dest='instance' )
parser.add_argument( '-v', dest='value' )
parser.add_argument( '-u', dest='units' )

args = parser.parse_args()

# Open the database
cache_db.open_db()

# Write to the database
cache_db.save_value_and_units( args.view, args.facility, args.instance, args.value, args.units )

File: cache/write_cache.php
<?php
  // Copyright 2018 BACnet Gateway.  All rights reserved.
  require_once $_SERVER["DOCUMENT_ROOT"] . "/../common/util.php";

  error_log( "==> post=" . print_r( $_POST, true ) );

  if (
    isset( $_POST['view'] ) &&
    isset( $_POST['facility'] ) &&
    isset( $_POST['instance'] ) &&
    isset( $_POST['value'] ) &&
    isset( $_POST['units'] )
    )
  {
    // Format command
    error_log( '==========> PHP_OS=' . PHP_OS . ' PHP_OS_FAMILY=' . PHP_OS_FAMILY );
    $sudo = ( PHP_OS_FAMILY == 'Linux' ) ? 'sudo ' : '';
    error_log( '====> sudo=' . $sudo );
    $command = $sudo . quote( getenv( 'PYTHON' ) ) . ' write_cache.py 2>&1'
      . ' -w ' . quote( $_POST['view'] )
      . ' -f ' . quote( $_POST['facility'] )
      . ' -i ' . quote( $_POST['instance'] )
      . ' -v ' . quote( $_POST['value'] )
      . ' -u ' . quote( $_POST['units'] );

    // Execute command
    error_log( '==> command=' . $command );
    exec( $command, $output, $status );
    error_log( '==> output=' . print_r( $output, true ) );
  }

  echo json_encode( 'done' );
?>
